### PR TITLE
게시판 CRUD 구현 중 (~ing)

### PR DESCRIPTION
**Pull request by EunjungOh123**
작업 내용

☐ 레시피 등록 (CKeditor 사용하여 contents에 복수의 이미지 삽입, 로그인한 사용자만 작성 가능)
☐ 레시피 목록 (페이징 처리까지 ok, 목록 페이지는 추후 변경 예정?)
☐ 레시피 상세 페이지 (contents에 삽입된 이미지 불러오기 성공)

### CODE REVIEW

**Comment by EunjungOh123** [comment]
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[CKResponse.java](https://github.com/EunjungOh123/BakingIn-Project/blob/321e052fdea15016fc4dbd959b384a7f3d6af603/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fdto%2FCKResponse.java)|0%|:x:|
|[BoardDto.java](https://github.com/EunjungOh123/BakingIn-Project/blob/321e052fdea15016fc4dbd959b384a7f3d6af603/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fdto%2FBoardDto.java)|0%|:x:|
|[BoardController.java](https://github.com/EunjungOh123/BakingIn-Project/blob/321e052fdea15016fc4dbd959b384a7f3d6af603/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fcontroller%2FBoardController.java)|0%|:x:|
|[BoardRestController.java](https://github.com/EunjungOh123/BakingIn-Project/blob/321e052fdea15016fc4dbd959b384a7f3d6af603/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fcontroller%2FBoardRestController.java)|0%|:x:|
|[BoardException.java](https://github.com/EunjungOh123/BakingIn-Project/blob/321e052fdea15016fc4dbd959b384a7f3d6af603/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fexception%2FBoardException.java)|0%|:x:|
|[BoardErrorCode.java](https://github.com/EunjungOh123/BakingIn-Project/blob/321e052fdea15016fc4dbd959b384a7f3d6af603/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Ftype%2FBoardErrorCode.java)|0%|:x:|
|[BoardServiceImpl.java](https://github.com/EunjungOh123/BakingIn-Project/blob/321e052fdea15016fc4dbd959b384a7f3d6af603/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fservice%2FBoardServiceImpl.java)|0%|:x:|
|[Board.java](https://github.com/EunjungOh123/BakingIn-Project/blob/321e052fdea15016fc4dbd959b384a7f3d6af603/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fentity%2FBoard.java)|0%|:x:|

|Total Project Coverage|0%|:x:|
|:-|:-:|:-:|

**Comment by EunjungOh123** [comment]
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[CKResponse.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fdto%2FCKResponse.java)|0%|:x:|
|[BoardDto.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fdto%2FBoardDto.java)|0%|:x:|
|[AdminCategoryController.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fcontroller%2FAdminCategoryController.java)|0%|:x:|
|[BoardController.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fcontroller%2FBoardController.java)|0%|:x:|
|[BoardRestController.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fcontroller%2FBoardRestController.java)|0%|:x:|
|[BoardException.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fexception%2FBoardException.java)|0%|:x:|
|[BoardErrorCode.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Ftype%2FBoardErrorCode.java)|0%|:x:|
|[CategoryException.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fexception%2FCategoryException.java)|0%|:x:|
|[CategoryErrorCode.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fexception%2FCategoryErrorCode.java)|0%|:x:|
|[GlobalExceptionHandler.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fexception%2FGlobalExceptionHandler.java)|0%|:x:|
|[BoardServiceImpl.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fservice%2FBoardServiceImpl.java)|0%|:x:|
|[Board.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fentity%2FBoard.java)|0%|:x:|
|[RecipeCategoryServiceImpl.java](https://github.com/EunjungOh123/BakingIn-Project/blob/1b9c5514f0027e2287e559be6c9d945980ab7f99/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fcategory%2Fservice%2FRecipeCategoryServiceImpl.java)|0%|:x:|

|Total Project Coverage|0%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/zerobase/bakingin_project/board/controller/BoardController.java [hunk 1]
```diff
@@ -1,42 +1,69 @@
...
-        return "/board/add";
+        return "redirect:/board/list";
     }
 
+    @GetMapping("/list")
+    public String list(Model model, @PageableDefault(size = 10, sort = "id", direction = Sort.Direction.DESC) Pageable pageable) {
+        Page<Board> list = boardService.boardList(pageable);
+
+        int nowPage = list.getPageable().getPageNumber() + 1;
+        int startPage = Math.max(nowPage - 4, 1);
```
Comment (KimHyoJin) [inline src/main/java/com/zerobase/bakingin_project/board/controller/BoardController.java:53]: 페이지가 총 10개가 보여지는 것 같은데요, 요 부분의 요구사항이 변경된다했을때 (예 : 전체 서비스에서 페이지가 5개만 보이게 해주세요) 좀더 고쳐야할 부분을 쉽게 찾을 수있도록 따로 util성 메소드로 빼면 좋을 것같아요!

File: src/main/java/com/zerobase/bakingin_project/admin/category/service/RecipeCategoryServiceImpl.java [hunk 2]
```diff
@@ -29,22 +31,33 @@ public List<RecipeCategoryDto> list() {
...
     }
 
     @Override
     public void update(InputRecipeCategory inputRecipeCategory) {
-        Optional<RecipeCategory> optionalRecipeCategory = categoryRepository.findById(inputRecipeCategory.getId());
+        RecipeCategory category = categoryRepository.findById(inputRecipeCategory.getId())
+                .orElseThrow(() -> new CategoryException(CategoryErrorCode.CATEGORY_NOT_EXIST));
+
+        category.setCategoryName(inputRecipeCategory.getCategoryName());
+        category.setUsingYn(inputRecipeCategory.isUsingYn());
```
Comment (KimHyoJin) [inline src/main/java/com/zerobase/bakingin_project/admin/category/service/RecipeCategoryServiceImpl.java:58]: 뒤에 yn이라는 postfix가 붙는 것보단 필드를 using으로 이름을 하는 것은 어떨까요? Boolean값이기 때문에 t/f에 해당하는 값이 들어감은 쉽게 인지가 가능할것 같아요.

File: src/main/java/com/zerobase/bakingin_project/board/dto/BoardDto.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                .summary(board.getSummary())
+                .ingredient(board.getIngredient())
+                .contents(board.getContents())
+                .createAt(board.getCreateAt())
+                .views(board.getViews())
+                .build();
+    }
+
+    public String getCreateAtText() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy.MM.dd HH:mm");
```
Comment (KimHyoJin) [inline src/main/java/com/zerobase/bakingin_project/board/dto/BoardDto.java]: 매번 DateTimeFormatter를 만들지 않고 const로 빼줘도 좋을 것 같습니다

File: src/main/java/com/zerobase/bakingin_project/admin/category/service/RecipeCategoryServiceImpl.java (hunk 2 above)
Comment (KimHyoJin) [reply src/main/java/com/zerobase/bakingin_project/admin/category/service/RecipeCategoryServiceImpl.java:58]: 그리고 요 값은 어떤역할을 하는 건가용? 언제 변경되는지도 궁금해요

**Review by KimHyoJin — APPROVED** [review]
안녕하새요 밍밍멘토입니다. 코드를 굉장히 가독성있게 잘 짜시네요. 코드읽기가 수월하여 리뷰하는데 큰 도움이 되었습니다. 몇가지 코멘트 드렸어요. 수고많으셨습니다

**Comment by EunjungOh123** [comment]
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[CKResponse.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fdto%2FCKResponse.java)|0%|:x:|
|[BoardDto.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fdto%2FBoardDto.java)|0%|:x:|
|[AdminCategoryController.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fcontroller%2FAdminCategoryController.java)|0%|:x:|
|[BoardController.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fcontroller%2FBoardController.java)|0%|:x:|
|[BoardRestController.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fcontroller%2FBoardRestController.java)|0%|:x:|
|[BoardException.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fexception%2FBoardException.java)|0%|:x:|
|[BoardErrorCode.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Ftype%2FBoardErrorCode.java)|0%|:x:|
|[CategoryException.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fexception%2FCategoryException.java)|0%|:x:|
|[CategoryErrorCode.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fexception%2FCategoryErrorCode.java)|0%|:x:|
|[GlobalExceptionHandler.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fexception%2FGlobalExceptionHandler.java)|0%|:x:|
|[BoardServiceImpl.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fservice%2FBoardServiceImpl.java)|0%|:x:|
|[Board.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fboard%2Fentity%2FBoard.java)|0%|:x:|
|[RecipeCategoryServiceImpl.java](https://github.com/EunjungOh123/BakingIn-Project/blob/4dc83e6099f2ce04524a61044fc5e75eeec9ad26/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Fbakingin_project%2Fadmin%2Fcategory%2Fservice%2FRecipeCategoryServiceImpl.java)|0%|:x:|

|Total Project Coverage|0%|:x:|
|:-|:-:|:-:|

**Pull request merged**